### ISSUE TITLE
Backgrounds don't become unpinned if you buy a set containing them

### ISSUE THREAD

**Issue by shanaqui** [HabitRPG/habitica#13287]
If you pin a unique item that can only be purchased once, so that it appears in your rewards column, purchasing it should unpin it from your rewards column.

In the case of backgrounds, there are two ways to purchase them: separately, or a set. If you pin a background and then purchase it on its own, then it's unpinned correctly. If you pin a background, and then purchase a set containing that background, it doesn't get unpinned. It then appears to be purchasable from the rewards column.

If you attempt that, you get a message saying the purchase was successful, but no gems are taken and the item is not unpinned.

The ideal fix for this is to make purchasing a set of backgrounds unpin any backgrounds that are in that set.

**Comment by j-dover** [comment]
Hi @shanaqui , I am interested in working on this issue.

**Comment by Alys** [comment]
@j-dover Thanks! I've marked this as in progress for you. Post here if you have questions or run into trouble.

**Comment by shanaqui** [comment]
@j-dover Hi! Are you still interested in working on this issue? If not, or if we don't hear back in a week, I'll mark it as "help wanted" again so folks can see it's available. Thanks!

**Comment by j-dover** [comment]
@shanaqui Hi! I'm still working on the issue, and I found its cause. The unlock() function from website/common/script/ops/unlock.js does not use removeItemByPath() to unpin items when there is a full set. 

I implemented a foreach loop to unpin each pinned purchased item from the full set. I was able to get pinned backgrounds to unpin from the Backgrounds menu and the Rewards column when purchased from a set during manual testing. It works with 1-3 pinned backgrounds when purchasing a full set. 

My code passed the Full Test suite and the individual test suite for the Common folder. Can I commit and push my code?
